tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs

```diff
@@ -89,5 +94,208 @@ public void Initialize()
             // All mocks are consumed
             Assert.AreEqual(0, HttpMessageHandlerFactory.MockHandlersCount());
         }
+
+        [TestMethod]
+        [Description("Provide only username of Federated user. Result in integrated auth call with no password xml tag.")]
+        public async Task ProvideUsernameOfFederatedUser_NoPasswordXmlTagTestAsync()
```

Comment:
rename the test to what it is and what it does. say, something like IntegratedAuthUsingUpn_GetsTokenFromService()

Reply:
done.